homeassistant/components/mqtt_eventstream.py

```diff
@@ -0,0 +1,93 @@
+"""
+homeassistant.components.mqtt_eventstream
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Connect multiple Home Assistant instances via mqtt.
+
+Key difference from using the "remote" component:
+* The Home Assistant instance(s) may be behind a firewall that
+  will not route HTTP traffic to it from external networks.
+* If a subscribing Home Assistant instance is unavailable, messages
+  for it will buffer on the MQTT queue for later delivery.
+
+Configuration:
+
+To use the mqtt_eventstream component you will need to add the component to
+your configuration.yaml file.
+
+If you do not specify a publish_topic you will not forward events to the queue.
+If you do not specify a subscribe_topic then you will not receive events from
+the remote server.
+
+mqtt_eventstream:
+  publish_topic: MyServerName
+  subscribe_topic: OtherHaServerName
+"""
+import json
+from homeassistant.core import EventOrigin, State
+from homeassistant.const import (
+    MATCH_ALL, EVENT_TIME_CHANGED, EVENT_CALL_SERVICE,
+    EVENT_SERVICE_EXECUTED, EVENT_STATE_CHANGED)
+import homeassistant.loader as loader
+from homeassistant.remote import JSONEncoder
+
+DOMAIN = "mqtt_eventstream"
+DEPENDENCIES = ['mqtt']
+
+
+def setup(hass, config):
+    """ Setup our mqtt_eventstream component. """
+    def _event_handler(event):
+        """ Handle events by publishing them on the mqtt queue. """
+        if event.origin != EventOrigin.local:
+            return
+        if event.event_type in (
+                EVENT_TIME_CHANGED,
+                EVENT_CALL_SERVICE,
```

Comment:
Ah, right, I had meant to flag this for review or add a TODO.

Need to figure out the code for filtering out the service call made by the mqtt component publishing the event, otherwise you get into an infinite loop. I had filtered out all EVENT_CALL_SERVICE and EVENT_SERVICE_EXECUTED events to get around that for now.

I'll tag you after I do both that and verify the code manually with 2 instances.